src/Tasks/GenerateResource.cs

```diff
@@ -995,7 +995,8 @@ private bool TransformResourceFilesUsingResGen(List<ITaskItem> inputsToProcess,
 
             // We need to do a whole lot of work to make sure that we're not overrunning the command line ... UNLESS
             // we're running ResGen 4.0 or later, which supports response files. 
-            if (!_resgenPath.Equals(Path.GetDirectoryName(NativeMethodsShared.GetLongFilePath(ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("resgen.exe", TargetDotNetFrameworkVersion.Version35))), StringComparison.OrdinalIgnoreCase))
+            if (NativeMethodsShared.IsWindows &&
```

Comment:
The Mono's ResGen doesn't work with the GenerateResource task for more reasons than just the response files. It doesn't support the "/r:" reference arguments either. It would need significant changes to be used in this scenario in any case, thus adding NativeSharedMethods.IsMono condition doesn't make sense. Updating Mono ResGen would make sense. Also, I am not sure whether NativeSharedMethods.IsMono checks for Mono runtime or Mono toolchain, which may not match on Windows.

In my opinion the best option is to remove all the NativeSharedMethods.IsWindows conditions from both GenerateResource.cs and ResGen.cs. At the same time the disabled unit tests could be reenabled on OS X. The option to use custom ResGen tool was added with .NET 4 on Windows and thus any custom ResGen tool that was used had to support response files in the first place.

What I was trying to fix was invocation of our custom ResGen, which is based on Mono's ResGen. Any other case was already broken and remains broken even after applying the patch. I tried to go with the least invasive way, by enabling the command line splitting, where the conditions were not aligned over the code. However I am equally fine with aligning the conditions the other way and allowing response files on all platforms.